api/v1alpha1/gateway_types.go

```diff
@@ -97,7 +127,7 @@ type Listener struct {
 	//
 	// Support:
 	// +optional
-	TLS *ListenerTLS `json:"tls,omitempty"`
+	TLS *TLSConfig `json:"tls,omitempty"`
```

Comment:
Maybe we need to step back from attaching TLS to the listener. The `HTTPRouteHost` has the hostname, maybe that's the right place to put a TLS configuration? That would strongly bind TLS configuration and virtual hosts in a way that's currently only implicit (even with the ListenerCertificates changes below).

Reply:
@jpeach IMO we need to get https://github.com/kubernetes-sigs/service-apis/issues/95 resolved to move forward with this or any of the other PR's that address TLS.

Reply:
If the App Dev is responsible for managing TLS config, then I agree that `xRoute` is the right place for all TLS config management.